src/CliCore.ts

```diff
@@ -106,6 +107,7 @@ export default class CliCore {
         const contextData = contextFilePath ? await this.readFileAndParse(contextFilePath, importPath) : {};
         options.importPath = importPath; //path is where local imports will be sourced from. We sneak path in with the options
         this.templateProcessor = new TemplateProcessor(input, contextData, options);
+        if (this.onInit) this.templateProcessor.onInit = this.onInit;
```

Comment:
This is a pretty weird line. I am not sure validation logic should be out here. Shouldn't the TemplateProcessor simply ignore onInit that is undefined or null?